processor/spanmetricsprocessor/processor.go

```diff
@@ -377,22 +433,28 @@ func (p *processorImp) aggregateMetrics(traces pdata.Traces) {
 			continue
 		}
 		serviceName := attr.StringVal()
-		p.aggregateMetricsForServiceSpans(rspans, serviceName)
+		p.aggregateMetricsForServiceSpans(ctx, rspans, serviceName)
 	}
+
+	stats.Record(ctx, dimensionsCacheSize.M(int64(p.metricKeyToDimensions.Len())))
+	stats.Record(ctx, uniqueTimeSeriesCount.M(int64(len(p.callSum))))
 }
 
-func (p *processorImp) aggregateMetricsForServiceSpans(rspans pdata.ResourceSpans, serviceName string) {
+func (p *processorImp) aggregateMetricsForServiceSpans(ctx context.Context, rspans pdata.ResourceSpans, serviceName string) {
 	ilsSlice := rspans.InstrumentationLibrarySpans()
 	for j := 0; j < ilsSlice.Len(); j++ {
 		ils := ilsSlice.At(j)
 		spans := ils.Spans()
+		spanCounter := 0
 		for k := 0; k < spans.Len(); k++ {
 			span := spans.At(k)
 			if filterspan.SkipSpan(p.include, p.exclude, span, rspans.Resource(), ils.InstrumentationLibrary()) {
 				continue
 			}
 			p.aggregateMetricsForSpan(serviceName, span, rspans.Resource().Attributes())
+			spanCounter++
 		}
+		stats.Record(ctx, spanProcessedCount.M(int64(spanCounter)))
```

Comment:
This is still inside a loop

Reply:
I moved this counter one loop out, but it's still in the `aggregateMetricsForServiceSpans` function. I don't think we can move it out any further since there is `filterspan.SkipSpan` with in the loop in `aggregateMetricsForServiceSpans` that filters out spans and we need to account for that in our span processed count.